### PR TITLE
Bugfix/prod 1477 dbfs filesize wait

### PR DESCRIPTION
**Pull request by gorskysd**
On Databricks Azure (possibly AWS too, but hasn't been observed), event log files being uploaded to DBFS may first appear with zero file size, then some time later become populated with data. Prior to this PR, the library would identify the zero-file-size files and upload them before they were actually populated with data.

This PR fixes the issue by doing to things:
1. Continues polling for new event log files while _any_ of the files have zero file size
2. Continues polling for new event log files if the data size changes between two consecutive polling periods

Jira Ticket: https://synccomputing.atlassian.net/browse/PROD-1477

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -19,3 +19,9 @@ endif
 .PHONY: tidy
 tidy: format lint
 
+# Removes the directory that contains bytecode cache files
+# that are automatically generated by python.
+.PHONY: clean
+clean:
+	find . -type f -name "*.pyc" | xargs rm -fr
+	find . -type d -name __pycache__ | xargs rm -fr
```
Comment (gorskysd) [inline Makefile:27]: This is convenience function for developers. Run `make clean` to delete pythons cache files.

**Review by brandon-kaplan — DISMISSED** [review]

File: sync/_databricks.py [hunk 2]
```diff
@@ -1725,6 +1725,25 @@ def _dbfs_directory_has_all_rollover_logs(contents: dict, run_end_time_millis: f
     )
 
 
+def _dbfs_any_file_has_zero_size(dbfs_contents: Dict):
```
Comment (aditya-jaishankar) [inline sync/_databricks.py:1728]: Super nitpicky but perhaps you also wanted a return type hint `-> bool | None`

**Review by aditya-jaishankar — DISMISSED** [review]
Nice! Very clean way to implement the partial file checks.

File: sync/_databricks.py (hunk 2 above)
Comment (gorskysd) [reply sync/_databricks.py]: good catch. Type hints are important, definitely not a nit :)

**Review by aditya-jaishankar — APPROVED** [review]
LGTM!

**Pull request merged**